datastore-api/app/core/kgs/connector.py

```diff
@@ -362,7 +389,65 @@ async def get_object_by_id_msearch(self, kg_name, ids):
                 logger.info("Not FOUND")
         return objs
 
-    async def extract_subgraph(self, kg_name, nids: List[str], hops=2):
+
+
+    async def extract_extra_nodes(self, kg_name, node_ids):
+        cross_combined_ids = []
+        for s in node_ids:
+            for t in node_ids:
+                cross_combined_ids.append([s, t])
+
+        all_edges = await self.get_edge_msearch(kg_name, cross_combined_ids)
+
+        return all_edges
```

Comment:
Please annotate this method. The method name is also misleading: It says "extract nodes" but it return edges